src/Translator.php

```diff
@@ -0,0 +1,542 @@
+        foreach ($this->modulePaths as $modulePath) {
+            $jsPath = $this->getJSLangDirectories($modulePath);
+            foreach ((array) $jsPath as $langDir) {
+                foreach (glob($langDir . '/src/*.json') as $path) {
+                    $str = file_get_contents($path);
+                    $this->originalJson[$path] = $this->jsonDecode($str);
+                }
+            }
+        }
+        $this->log('Finished backing up ' . count($this->originalJson) . ' json files');
+    }
+
+    /**
+     * Backup local yaml files in memory prior to replacing local copies with transifex
+     */
+    private function storeYaml(): void
+    {
+        $this->log('Backing up local yaml files');
+        $this->originalYaml = [];
+        foreach ($this->modulePaths as $modulePath) {
+            foreach (glob($this->getLangDirectory($modulePath) . '/*.yml') as $path) {
+                $rawYaml = file_get_contents($path);
+                $this->originalYaml[$path] = Yaml::parse($rawYaml);
+            }
+        }
+        $this->log('Finished backing up ' . count($this->originalYaml) . ' yaml files');
+    }
+
+    /**
+     * Set mtime to a year ago so that transifex will see these as obsolete
+     */
+    private function setJsonAndYmlFileTimes()
+    {
+        $date = date('YmdHi.s', strtotime('-1 year'));
+        foreach ($this->modulePaths as $modulePath) {
+            $name = $this->getModuleName($modulePath);
+            $this->log("Pulling sources from transifex for <info>{$name}</info> (min %{$this->txMinimumPerc} delta)");
+            $ymlLang = $this->getLangDirectory($modulePath);
+            if ($ymlLang) {
+                $this->exec("find $ymlLang -type f \( -name \"*.yml\" \) -exec touch -t $date {} \;");
+            }
+            foreach ($this->getJSLangDirectories($modulePath) as $jsLangDir) {
+                $this->exec("find $jsLangDir -type f \( -name \"*.json*\" \) -exec touch -t $date {} \;");
+            }
+        }
+    }
+
+    /**
+     * Update sources from transifex
+     */
+    private function transifexPullSource()
+    {
+        foreach ($this->modulePaths as $modulePath) {
+            // ensure .tx/config is up to date
+            $contents = file_get_contents("$modulePath/.tx/config");
+            if (strpos($contents, '[o:') === false) {
+                $this->exec('tx migrate', $modulePath);
+                // delete .bak files created as part of tx migrate
+                foreach (scandir("$modulePath/.tx") as $filename) {
+                    if (pathinfo($filename, PATHINFO_EXTENSION) !== 'bak') {
+                        continue;
+                    }
+                    $this->log("Deleting $modulePath/.tx/$filename");
+                    unlink("$modulePath/.tx/$filename");
+                }
+            }
+            // pull from transifex
+            $this->exec("tx pull -a -s -t -f --minimum-perc={$this->txMinimumPerc}", $modulePath);
```

Comment:
https://developers.transifex.com/docs/cli
"Migrating from older versions of the client"

Reply:
I explicitly asked for specific information about the specific changes you've actually made, because you've already had to parse the information in those docs. That's a pretty long document and it's for general cases - it doesn't explain why _you_ have made the choices you've made, and changed specifically what you've changed.
Can you please explain the specific changes you've made?

Reply:
- There's a new version of tx that uses go that we use in this module
- Cow used the deprecation version of tx that uses python
- The new version expects a different syntax for `.tx/config`
- Running `tx migrate` updates the syntax in `.tx/config` for you - e.g. adds the `[o:`